### ISSUE TITLE
Failure to install on Travis CI

### ISSUE THREAD

**Issue by dcvz** [tuist/tuist#501]
**Describe the bug**
The installation script fails when trying to install tuist in TravisCI.

First issue was along fetching the release:
`-e:166:in '<main>': undefined method 'detect' for nil:NilClass (NoMethodError)`

_We managed to get past this by forking the install script and directly stating the zip download url._

Second issue was along moving the directories:
`mv: rename /var/folders/17/5mc7816d3mndxjqgplq6057w0000gn/T/d20190918-8275-162s8gi/tuistenv/tuistenv to /usr/local/bin/tuist: No such file or directory`

**To Reproduce**
Setup a travis CI config script that looks something similar to:
```
language: swift
os: osx
osx_image: xcode10.3

before_install:
  - /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/tuist/tuist/master/install/install)"
before_script:
  - tuist generate

xcode_workspace: My.xcworkspace
xcode_scheme: MyScheme
xcode_destination: platorm=iOS Simulator,OS=12.4,name=iPhone X
```

**Temporary Fix**
We went ahead and fixed the issue by using brew to install tuist like so:
```bash
brew install https://url_to_formula.rb
```

**Comment by dcvz** [comment]
I can document this install method if ya'll are ok with it.

**Comment by pepibumur** [comment]
Thanks for reporting this @dcvz. I'll look into it.

**Comment by dcvz** [comment]
**Update:** After more investigation it seems to be that requests to Github are failing. As even after I managed to install Tuist I'm getting an error on `generate`:

```
Error: Error decoding JSON from API: The data couldn’t be read because it isn’t in the correct format.
```

I'm starting to think that perhaps Travis is hitting some request limit or something.. I will keep investigating.

But separate question @pepibumur. Why is `generate` invoking the Github API?

**Comment by pepibumur** [comment]
> I'm starting to think that perhaps Travis is hitting some request limit or something.. I will keep investigating.

I experienced something similar yesterday so most likely. I'm thinking that we could perhaps put the installation script somewhere else, like a Google Cloud Storage bucket.

> But separate question @pepibumur. Why is generate invoking the Github API?

So basically, what you install in your system is not `tuist`, but `tuistenv`. Tuist env serves as a gateway to ensure that everyone in the team uses the same version of Tuist. When you run it for the first time, it pulls the latest version and places it under `~/.tuist/Versions/x.y.z/xxx`. 

We ruled out the idea of using Homebrew because I've had several experiences in the past where developers were using different versions of `swiftlint` that resulted in different results. Until Swift provides a mechanism for installing global packages in the environment, I wanted the experience of Tuist being reproducible.

Unfortunately the approach that we followed using GitHub's API and releases doesn't seem to be a good idea, so we might need to put those assets somewhere else.
